src/matlab2tikz.m

```diff
@@ -1770,6 +1771,12 @@ function assertRegularAxes(handle)
 
         userWarning(m2t, 'Automatically added label ''%s'' for line plot.', labelName);
         generatedCodeSoFar = [generatedCodeSoFar, labelCode];
+        
+        if ~isempty(lineTag)
+            labelName = sprintf('%s', lineTag);
```

Comment:
Also i think, that automaticLabels and special tags are rather exclusive, so i would check for an label first and in the else clause add the automatic label